dashboard/client/src/pages/log/Logs.tsx

```diff
@@ -25,126 +34,233 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-const useLogs = () => {
-  const { search: urlSearch } = useLocation();
-  const { host, path } = useParams();
-  const searchMap = new URLSearchParams(urlSearch);
-  const urlFileName = searchMap.get("fileName");
-  const [origin, setOrigin] = useState<string>();
-  const [fileName, setFileName] = useState(searchMap.get("fileName") || "");
-  const [log, setLogs] =
-    useState<undefined | string | { [key: string]: string }[]>();
-  const [downloadUrl, setDownloadUrl] = useState<string>();
-
-  useEffect(() => {
-    setFileName(urlFileName || "");
-  }, [urlFileName]);
-
-  useEffect(() => {
-    let url = "log_index";
-    setLogs("Loading...");
-    if (host) {
-      url = decodeURIComponent(host);
-      setOrigin(new URL(url).origin);
-      if (path) {
-        url += decodeURIComponent(path);
-      }
-    } else {
-      setOrigin(undefined);
-    }
-    setDownloadUrl(getLogDownloadUrl(url));
-    getLogDetail(url)
-      .then((res) => {
-        if (res) {
-          setLogs(res);
-        } else {
-          setLogs("(This file is empty.)");
-        }
-      })
-      .catch(() => {
-        setLogs("Failed to load");
-      });
-  }, [host, path]);
-
-  return {
-    log,
-    origin,
-    downloadUrl,
-    host,
-    path,
-    fileName,
-    setFileName,
-  };
+export const StateApiLogsListPage = () => {
+  const classes = useStyles();
+  const [searchParams] = useSearchParams();
+  const nodeId = searchParams.get("nodeId");
+  const folder = searchParams.get("folder");
+  const [fileName, setFileName] = useState("");
+
+  const backFolder = folder
+    ? [...folder.split("/").slice(0, -1)].join("/")
+    : undefined;
+  const backHref =
+    // backGlob is undefined when glob is empty
+    // backGlob is empty string when glob is 1 level deep.
+    backFolder !== undefined && nodeId
+      ? `/logs/?nodeId=${encodeURIComponent(
+          nodeId,
+        )}&folder=${encodeURIComponent(backFolder)}`
+      : `/logs/`;
+
+  return (
+    <div className={classes.root}>
+      <TitleCard title="Logs Viewer">
+        <Paper>
+          {!nodeId && <p>Select a node to view logs</p>}
+          {nodeId && (
+            <React.Fragment>
+              <p>Node: {nodeId}</p>
+              <p>{decodeURIComponent(folder || "")}</p>
+            </React.Fragment>
+          )}
+          {nodeId && (
+            <div>
+              <Button
+                component={Link}
+                variant="contained"
+                to={backHref}
+                className={classes.search}
+              >
+                Back To ../
+              </Button>
+              <SearchInput
+                defaultValue={fileName}
+                label="File Name"
```

Comment:
Btw, this doesn't seem to work when I used it

Reply:
oh shoot, I didn't test this. let me test thi

Reply:
Fixed!